data/transactions/logic/assembler.go

```diff
@@ -1872,22 +1954,32 @@ func (ops *OpStream) trace(format string, args ...interface{}) {
 	fmt.Fprintf(ops.Trace, format, args...)
 }
 
-func (ops *OpStream) typeErrorf(format string, args ...interface{}) {
+func (ops *OpStream) typeErrorf(opcode token, format string, args ...interface{}) {
 	if ops.typeTracking {
-		ops.errorf(format, args...)
+		ops.record(opcode.errorf(format, args...))
 	}
 }
 
+func reJoin(instruction string, args []token) string {
+	var buf bytes.Buffer
+	buf.WriteString(instruction)
+	for _, t := range args {
+		buf.WriteString(" ")
+		buf.WriteString(t.str)
+	}
+	return buf.String()
+}
+
```

Comment:
i tried to make this better in some way but I failed

Reply:
It's a little dopey, but it's only used in error messages, so I don't mind too much. We _could_ use the column information in the tokens to do a better job of reJoining (so that the amount of space between tokens is restored), but ain't nobody got time for that.